docs/upgrading.md

```diff
@@ -0,0 +1,111 @@
+
+## Terminology
+
+There are currently two implementations of Clear Containers:
+
+- Clear Containers 2.x (or just 2.x) refers to the previous implementation
+  of Clear Containers which uses a runtime called `cc-oci-runtime` and
+  whose code is available at: https://github.com/01org/cc-oci-runtime
+
+- Clear Container 3.x (or just 3.x) refers to the current implementation
+  which uses a runtime called `cc-runtime` and whose code is available
+  at: https://github.com/clearcontainers/runtime
+
+## Maintenance warning
+
+The 2.x codebase is no longer being developed. New releases will _only_ be
+considered for significant bugs.
+
+The main development focus is now on the 3.x branch. As such, all 2.x
+users are encouraged to switch to the 3.x implementation.
+
+## Introduction
+
+The two Clear Containers implementations share certain architectural
+elements. Namely, both use a shim process called `cc-shim` and a proxy
+process called `cc-proxy`. However, the 2.x versions of these
+elements *are incompatible* with their 3.x counterparts.
+
+## Supported operations and configurations
+
+### Systems using pre-built packages
+
+For operating systems that pre-built packages are provided for (such as
+Fedora and Ubuntu), upgrading is possible with some caveats:
+
+- Configuration files are not migrated (see [Configuration files](#configuration-files)).
+- It is necessary to perform a few cleanup steps as outlined in the
+  subsections below.
+
+Once the cleanup steps have been performed, simply follow the 3.x
+installation guides and any existing packaged versions of the 2.x
+components will be removed and the new 3.x components installed in their
+place.
+
+#### Upgrading an Ubuntu system
+
+Assuming the system was installed by following the [CC 2.x Ubuntu installation guide](https://github.com/01org/cc-oci-runtime/blob/master/documentation/Installing-Clear-Containers-on-Ubuntu.md):
+
+1. Perform the following cleanup steps:
+   ```bash
+   $ sudo rm -f /etc/systemd/system/docker.service.d/clr-containers.conf
+   $ sudo apt-get purge -y docker-engine
+   $ sudo rm -f /etc/apt/sources.list.d/cc-oci-runtime.list
+   ```
+
+1. Follow the [CC 3.x Ubuntu installation guide](docs/ubuntu-installation-guide.md).
+
+#### Upgrading a Fedora system
+
+Assuming the system was installed by following the [CC 2.x Fedora installation guide](https://github.com/01org/cc-oci-runtime/blob/master/documentation/Installing-Clear-Containers-on-Fedora.md):
+
+1. Perform the following cleanup steps:
+   ```bash
+   $ sudo rm -f /etc/systemd/system/docker.service.d/clr-containers.conf
+   $ sudo rm -f /etc/yum.repos.d/home:clearlinux:preview:clear-containers-2.1.repo
+   ```
+
+1. Follow the [CC 3.x Fedora installation guide](docs/fedora-installation-guide.md).
+
+## Unsupported operations and configurations
+
+### CentOS
+
+Since the CentOS installation instructions require the user to manually
+build and install various packages, upgrading is not supported on this
```

Comment:
Rather than say 'not supported', maybe we can say 'must be performed by hand' (nee: 'left as an exercise for the user' ;-)). That is, it is not that you cannot upgrade, it is just there is no nicely defined procedure to do so.

Reply:
Glad you raised this because there are 2 issues here: upgrading the system, but also whether the resulting system should be considered "supported" (in that we will consider bugs raised on such a system). I agree that we don't want to specify how to upgrade a CentOS system, but the question comes down to, "what if someone does it?" If they do, I think we say "not supported" but what do you think @sameo?

Reply:
I think this is the only remaining issue blocking this landing. @sameo - do you have a view on this?

Reply:
@jodh-intel I'd rather avoid saying "not supported" because we're not at a point where we can ignore users because they've done a manual upgrade. Realistically, if someone reports a bug on an upgraded CentOS we'll have to look at it I'm afraid.

Reply:
Hi @sameo, @grahamwhaley - I've updated the doc and moved the CentOS section below "Supported" -> "Not Recommended" which will hopefully make it clear ;)

Can you re-review?